src/adversarial/neurify.jl

```diff
@@ -70,47 +86,67 @@ function check_inclusion(reach::SymbolicInterval{HPolytope{N}}, output::Abstract
     # The output constraint is in the form A*x < b
     # We try to maximize output constraint to find a violated case, or to verify the inclusion, 
     # suppose the output is [1, 0, -1] * x < 2, Then we are maximizing reach.Up[1] * 1 + reach.Low[3] * (-1) 
-
+    
+    # x = model[:x]
     reach_lc = reach.interval.constraints
     output_lc = output.constraints
-
     n = size(reach_lc, 1)
     m = size(reach_lc[1].a, 1)
     model =Model(with_optimizer(GLPK.Optimizer))
     @variable(model, x[1:m])
     @constraint(model, [i in 1:n], reach_lc[i].a' * x <= reach_lc[i].b)
 
-    for i in 1:size(output.constraints, 1)
+
+    for i in 1:size(output_lc, 1)
         obj = zeros(size(reach.Low, 2))
         for j in 1:size(reach.Low, 1)
-            if output.constraints[i].a[j] > 0
-                obj += output.constraints[i].a[j] * reach.Up[j,:]
+            if output_lc[i].a[j] > 0
+                obj += output_lc[i].a[j] * reach.Up[j,:]
             else
-                obj += output.constraints[i].a[j] * reach.Low[j,:]
+                obj += output_lc[i].a[j] * reach.Low[j,:]
             end
         end
-        @objective(model, Min, obj' * [x; [1]])
+        @objective(model, Max, obj' * [x; [1]])
         optimize!(model)
-        # println("termination_status")
-        # println(termination_status(model))        
-        y = compute_output(nnet, value(x))
-        if !∈(y, output)
+        if termination_status(model) == MOI.OPTIMAL
+            y = compute_output(nnet, value(x))
+            if !∈(y, output)
+                if ∈(value(x), reach.interval)
+                    return CounterExampleResult(:violated, value(x))
+                else
+                    print("OPTIMAL, but x not in the input set")
+                    exit()
+                end
+            end
+            if objective_value(model) > output_lc[i].b
+                # println(objective_value(model)," ", output_lc[i].b)
+                return BasicResult(:unknown)
+            end
+        else
             if ∈(value(x), reach.interval)
-                return CounterExampleResult(:violated, value(x))
+                print("Not OPTIMAL, but x in the input set")
+                exit()
             end
-        end
-        if objective_value(model) > output.constraints[i].b
             return BasicResult(:unknown)
         end
         
     end
     return BasicResult(:holds)
 end
 
+function printconvex(shape)
```

Comment:
Just for debugging, right? I don't mind keeping it around, but if you don't think you'll need it again, we can just go ahead and delete it.

Reply:
Will remove.